crates/store/re_video/src/time.rs

```diff
@@ -28,50 +28,67 @@ impl Time {
         Self(v)
     }
 
+    /// `time_base` specifies the
     #[inline]
-    pub fn from_secs(v: f64, timescale: Timescale) -> Self {
-        Self((v * timescale.0 as f64).round() as i64)
+    pub fn from_secs_since_start(
+        secs_since_start: f64,
+        timescale: Timescale,
+        start_time: Self,
+    ) -> Self {
+        Self((secs_since_start * timescale.0 as f64).round() as i64 + start_time.0)
     }
 
     #[inline]
-    pub fn from_millis(v: f64, timescale: Timescale) -> Self {
-        Self::from_secs(v / 1e3, timescale)
+    pub fn from_millis_since_start(
+        millis_since_start: f64,
+        timescale: Timescale,
+        start_time: Self,
+    ) -> Self {
+        Self::from_secs_since_start(millis_since_start / 1e3, timescale, start_time)
     }
 
     #[inline]
-    pub fn from_micros(v: f64, timescale: Timescale) -> Self {
-        Self::from_secs(v / 1e6, timescale)
+    pub fn from_micros_since_start(
+        micros_since_start: f64,
+        timescale: Timescale,
+        start_time: Self,
+    ) -> Self {
+        Self::from_secs_since_start(micros_since_start / 1e6, timescale, start_time)
     }
 
     #[inline]
-    pub fn from_nanos(v: i64, timescale: Timescale) -> Self {
-        Self::from_secs(v as f64 / 1e9, timescale)
+    pub fn from_nanos_since_start(
```

Comment:
Maybe a single `fn from_duration_sicne_start(duration: std::time::Duration, …)` could replace all these, to produce a more minimal interface

Reply:
still have to keep a nanos version one because std::time::Duration has imho waaaay too much overhead for converting to our `VideoTimestamp` (we expect to converts hundereds of thousands in some cases)

Reply:
the other nastiness is that duration _panics_ when using negative values.

Actually that last one compells me strongly to roll back what I did. std::time::duration is not safe when dealing with user data